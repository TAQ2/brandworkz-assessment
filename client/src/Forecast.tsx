import React from "react";
import styled from "styled-components";
import {
  WiNightClear,
  WiDaySunny,
  WiRainMix,
  WiSnow,
  WiSleet,
  WiWindy,
  WiFog,
  WiCloudy
} from "react-icons/wi";
import { screenBreakpoints } from "./theme";

const WeatherIconContainer = styled.div`
  font-size: 20em;
  margin-right: 0.7em;
  display: flex;
  flex-direction: column;

  @media (max-width: ${screenBreakpoints.tablet}px) {
    font-size: 15em;
    margin-right: 0.5em;
  }

  @media (max-width: ${screenBreakpoints.small}px) {
    margin-right: 0em;
    margin-bottom: 0.2em;
  }
`;

const WeatherSummaryContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  @media (max-width: ${screenBreakpoints.small}px) {
    flex-direction: column;
  }
`;

const WeatherIcon: React.FC<{ type: string }> = ({ type }) => {
  switch (type) {
    case "clear-day":
      return <WiDaySunny />;
    case "clear-night":
      return <WiNightClear />;
    case "rain":
      return <WiRainMix />;
    case "snow":
      return <WiSnow />;
    case "sleet":
      return <WiSleet />;
    case "wind":
      return <WiWindy />;
    case "fog":
      return <WiFog />;
    case "cloudy":
      return <WiCloudy />;
    case "partly-cloudy-day":
      return <WiCloudy />;
    case "partly-cloudy-night":
      return <WiCloudy />;

    default:
      console.error(
        "renderWeatherIcon could not handle the weather type. it recieved: " +
          type
      );
      // @Incomplete - should render something more appropriate
      return <WiCloudy />;
  }
};

const Forecast: React.FC<{
  city: string;
  country: string;
  icon: string;
  temperature: number;
  humidity: number;
  windSpeed: number;
  summary: string;
}> = ({ city, country, icon, temperature, humidity, windSpeed, summary }) => {
  return (
    <div>
      <div style={{ textAlign: "center" }}>
        <div style={{ fontSize: "3em", fontWeight: "bold" }}>{city}</div>
        <div style={{ fontSize: "1.5em", color: "gray" }}>{country}</div>
      </div>
      <WeatherSummaryContainer>
        <WeatherIconContainer>
          <WeatherIcon type={icon} />
        </WeatherIconContainer>
        <div>
          <div style={{ fontWeight: "bold" }}>{summary}</div>
          <div>Temperature: {temperature + "\u00b0"}C</div>
          <div>Humidity: {humidity}%</div>
          <div>Wind speed: {windSpeed}mph</div>
        </div>
      </WeatherSummaryContainer>
    </div>
  );
};

export default Forecast;
