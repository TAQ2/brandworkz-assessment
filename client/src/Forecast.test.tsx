import React from "react";
import { shallow, mount } from "enzyme";

import Forecast from "./Forecast";

it("renders without crashing", () => {
  const forecastProps = {
    city: "test city",
    country: "test country",
    temperature: 111,
    humidity: 222,
    windSpeed: 333,
    summary: "test summar ",
    icon: "cloudy"
  };
  const wrapper = shallow(<Forecast {...forecastProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders all the props in the component", () => {
  const forecastProps = {
    city: "test city",
    country: "test country",
    temperature: 111,
    humidity: 222,
    windSpeed: 333,
    summary: "test summar "
  };

  // add icon separately because we do not render this prop
  const icon = "cloudy";
  const wrapper = mount(<Forecast {...forecastProps} icon={icon} />);

  Object.values(forecastProps).forEach(prop => {
    expect(wrapper.text().includes(String(prop))).toEqual(true);
  });
});
