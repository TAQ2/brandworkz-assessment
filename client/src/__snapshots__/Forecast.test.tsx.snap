// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "3em",
          "fontWeight": "bold",
        }
      }
    >
      test city
    </div>
    <div
      style={
        Object {
          "color": "gray",
          "fontSize": "1.5em",
        }
      }
    >
      test country
    </div>
  </div>
  <styled.div>
    <styled.div>
      <WeatherIcon
        type="cloudy"
      />
    </styled.div>
    <div>
      <div
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        test summar 
      </div>
      <div>
        Temperature: 
        111Â°
        C
      </div>
      <div>
        Humidity: 
        222
        %
      </div>
      <div>
        Wind speed: 
        333
        mph
      </div>
    </div>
  </styled.div>
</div>
`;
